src/app/helpers/transforms/transformEmbedded.test.ts

```diff
@@ -0,0 +1,57 @@
+import { MessageExtended } from '../../models/message';
+import { MailSettings, Api } from '../../models/utils';
+import { transformEmbedded } from './transformEmbedded';
+import { AttachmentData } from '../../models/attachment';
+import { prepareImages } from '../embedded/embeddedParser';
+import { SHOW_IMAGES } from 'proton-shared/lib/constants';
+
+const prepareImagesMock = prepareImages as jest.Mock;
+
+jest.mock('../embedded/embeddedFinder', () => ({
+    find: jest.fn(() => [])
+}));
+
+jest.mock('../embedded/embeddedParser', () => ({
+    mutateHTML: jest.fn(),
+    decrypt: jest.fn(),
+    prepareImages: jest.fn()
+}));
+
+const attachmentsCache = new Map<string, AttachmentData>();
+const api: Api = jest.fn();
+const cache: any = {};
+
+describe('transformEmbedded', () => {
+    describe('show', () => {
+        // Reference: Angular/test/specs/message/services/transformEmbedded.spec.js
+
+        const setup = async (message: MessageExtended = {}, mailSettings: MailSettings = {}) => {
+            await transformEmbedded(message, { mailSettings, attachmentsCache, api, cache });
+            return prepareImagesMock.mock.calls[0][1] as boolean;
+        };
+
+        afterEach(() => {
+            [prepareImagesMock].forEach((mock) => mock.mockClear());
+        });
+
+        it('should load embedded images when showEmbeddedImages = true', async () => {
+            const show = await setup({ showEmbeddedImages: true }, { ShowImages: 0 });
+            expect(show).toBe(true);
+        });
+
+        it('should load embedded images when showEmbedded = false and mailSettings = true', async () => {
+            const show = await setup({ showEmbeddedImages: false }, { ShowImages: SHOW_IMAGES.EMBEDDED });
+            expect(show).toBe(true);
+        });
+
+        it('should not load embedded images when showEmbedded = false and showImages = false', async () => {
+            const show = await setup({ showEmbeddedImages: false }, { ShowImages: 0 });
```

Comment:
Should there be a constant in `SHOW_IMAGES` with the value 0? Not sure what it represents

Reply:
The constants is in `proton-shared` and used for `proton-mail-settings`. 0 value is for NONE, it could be added to the enum I suppose. But to be honest, openning a PR for that is quite boring :p

Reply:
https://github.com/ProtonMail/proton-shared/pull/103 ;P
No, but really, I think it's a bit more clear if there is a constant `NONE = 0`